onyo/lib/onyo.py

```diff
@@ -1218,3 +1218,32 @@ def _rm_sanitize(self, paths: Iterable[Union[Path, str]]) -> list[Path]:
                                          '\n'.join(error_path_protected))
 
         return paths_to_rm
+
+    #
+    # UNSET
+    #
+    def unset(self, paths: Iterable[Union[Path, str]], values: str, yes: bool,
+              recursive: bool, depth: Union[int]) -> str:
+
+        # set and unset should select assets exactly the same way
+        assets_to_unset = self._set_sanitize(paths, recursive, depth)
+        keys = values.split(',')
+
+        if any([key in ["type", "make", "model", "serial"] for key in keys]):
+            log.error("Can't unset pseudo keys (name fields are required).")
+            raise ValueError("Can't unset pseudo keys (name fields are required).")
+
+        for asset in assets_to_unset:
+            contents = self._read_asset(asset)
+
+            for field in keys:
+                if field not in contents:
+                    # TODO: Should this error if field does not exist?
```

Comment:
No, it should not error. Otherwise, the user would /always/ need to use `filter` in order to constrain the asset list to only those which have the key.